File: 3105-longest-strictly-increasing-or-strictly-decreasing-subarray/3105-longest-strictly-increasing-or-strictly-decreasing-subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int n = nums.size(), increasing = std::numeric_limits<int>::min(), decreasing = std::numeric_limits<int>::max();
        int ans = 0, incCount = 0, decCount = 0;

        for (int i = 0; i < n; ++i){
            if (increasing < nums[i])
                ++incCount;
            
            else
                incCount = 1;

            if (decreasing > nums[i])
                ++decCount;
            else
                decCount = 1;

            increasing = nums[i], decreasing = nums[i];
            ans = max(ans, max(incCount, decCount));
        }

        return ans;
    }
};

File: 3105-longest-strictly-increasing-or-strictly-decreasing-subarray/3105-longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        n, ans = len(nums), 0
        increasing, decreasing = float('-inf'), float('inf')
        inc_count, dec_count = 0, 0

        for i in range(n):
            if increasing < nums[i]:
                inc_count += 1
            else:
                inc_count = 1
            increasing = nums[i]

            if decreasing > nums[i]:
                dec_count += 1
            else:
                dec_count = 1
            decreasing = nums[i]

            ans = max(ans, inc_count, dec_count)

        return ans